libsplinter/src/peer/mod.rs

```diff
@@ -1540,6 +1534,37 @@ fn log_connect_request_err(
     }
 }
 
+/// Check to make sure if multiple peer IDs point to the same endpoint, they are not diffferent
+/// Trust peer token. There can only be one Trust peer per endpoint, but multiple Challenge peers
+/// are okay.
+fn check_for_duplicate_endpoint(
+    peer_id: &PeerAuthorizationToken,
+    endpoints: &[String],
+    peer_map: &PeerMap,
+) -> bool {
+    #[cfg(feature = "challenge-authorization")]
+    if matches!(peer_id, PeerAuthorizationToken::Challenge { .. }) {
+        return false;
+    }
+
+    for endpoint in endpoints {
+        if let Some(peers) = peer_map.get_peer_from_endpoint(endpoint) {
+            for peer_meta in peers {
+                // allow irrefutable_let_patterns because it is required if challenge-authorization
+                // feature is enabled
+                #[allow(irrefutable_let_patterns)]
```

Comment:
could you just use the `matches!` macro like above?

Reply:
Yes, that does work, updated